ts/private/ts_project_worker.js

```diff
@@ -0,0 +1,384 @@
+const fs = require('fs')
```

Comment:
Was there a reason for cjs instead of es6? We're using es6 throughout the file...

Reply:
Haven't thought about it tbh. I guess that'd force users some node version and up? Nevertheless, I'll convert it to typescript when it gets stable enough